### PR TITLE
Remove filter locator.

### PR DESCRIPTION
**Pull request by ADmad**
I don't see much benefit of having `FilterLocatorInterface` and `FilterLocator`.

It's adds an unnecessary layer of abstraction and extra cognitive load for someone trying to understand how the plugin works.

It's highly unlikely that someone would want to change how filter loading works and even if they do, they can now override `FilterCollection::loadFilter()` in their custom collection class.

### CODE REVIEW

**Comment by dereuromark** [comment]
@burzum ?

**Comment by burzum** [comment]
@dereuromark yes? This is yet another change, at least not in the core, to cater the audience i don't care about. So we remove good OOP in favor of satisfying devs who can't or won't use their brain. So just do it.

No need to ask me for my opinion about these changes anymore. I pretty much gave up on CakePHP and experimenting with a frameworkless approach and other frameworks. https://github.com/prooph for example has very interesting things to offer.

File: src/Model/Filter/FilterCollectionInterface.php [hunk 1]
```diff
@@ -44,4 +44,15 @@ public function has($name);
...
+    /**
+     * Loads a search filter.
+     *
+     * @param string $name Filter name.
+     * @param string $filter Filter class name in short form like "Search.Value" or FQCN.
+     * @param array $options Filter options.
+     * @return \Search\Model\Filter\FilterInterface
+     * @throws \InvalidArgumentException When no filter was found.
+     */
+    public function loadFilter($name, $filter, array $options = []);
```
[the remark this answers is not in the document]
Comment (ADmad) [reply src/Model/Filter/FilterCollectionInterface.php]: @dereuromark Do you think the method should be renamed to just `load()`?

[the remark this answers is not in the document]
Comment (dereuromark) [reply src/Model/Filter/FilterCollectionInterface.php]: I think that sounds reasonable, given that we are inside a filter collection class.
But loadFilter() is not a big deal either.

[the remark this answers is not in the document]
Comment (ADmad) [reply src/Model/Filter/FilterCollectionInterface.php]: Okay then, keeping it unchanged :).

**Pull request merged**